File: Exercices/Séance 1 - C# vs Java-20231230/Ex1/Movie.cs
namespace be.ipl.domaine;

internal class Movie
{

    private string _title;
    private int _releaseYear;
    private List<Actor> _actors;
    private Director? _director = null;

    public Movie(String title, int releaseYear)
    {
        _actors = new List<Actor>();
        this._title = title;
        this._releaseYear = releaseYear;
    }

    // ? -> nullable
    public Director? Director
    {
        get { return _director; }
        set
        {
            if (value == null)
                return;

            this._director = value;
            _director?.AddMovie(this);

        }
    }

    // Properties with get and set 
    // Use of property 
    // Movie m = new Movie();
    // m.Title = "E.T";
    public string Title
    {
        get { return _title; }
        set { _title = value; }
    }


    public int ReleaseYear
    {
        get { return _releaseYear; }
        set { _releaseYear = value; }
    }


    public bool AddActor(Actor actor)
    {
        if (_actors.Contains(actor))
            return false;

        _actors.Add(actor);
        if (!actor.ContainsMovie(this))
            actor.AddMovie(this);

        return true;
    }

    public bool ContainsActor(Actor actor)
    {
        return _actors.Contains(actor);
    }

    public override string ToString()
    {
        return "Movie [title=" + _title + ", releaseYear=" + _releaseYear + "]";
    }

}

File: Exercices/Séance 1 - C# vs Java-20231230/ActorsDB/src/be/ipl/domaine/Movie.java
package be.ipl.domaine;

import java.util.ArrayList;
import java.util.List;

public class Movie {

	private String title;
	private int releaseYear;
	private List<Actor> actors;
	private Director director;
	
	public Movie(String title, int releaseYear) {
		actors = new ArrayList<Actor>();
		this.title = title;
		this.releaseYear = releaseYear;
	}

	public Director getDirector() {
		return director;
	}
	public void setDirector(Director director) {
		if (director == null)
			return;
		this.director = director;
		director.addMovie(this);
	}

	public String getTitle() {
		return title;
	}
	public void setTitle(String title) {
		this.title = title;
	}
	public int getReleaseYear() {
		return releaseYear;
	}
	public void setReleaseYear(int releaseYear) {
		this.releaseYear = releaseYear;
	}

	public boolean addActor(Actor actor){
		if (actors.contains(actor))
			return false;

		actors.add(actor);
		if (!actor.containsMovie(this))
			actor.addMovie(this);
		
		return true;
	}

	public boolean containsActor(Actor actor){
		return actors.contains(actor);
	}

	@Override
	public String toString() {
		return "Movie [title=" + title + ", releaseYear=" + releaseYear + "]";
	}
	
	
	
}
